bindings/zig/src/opendal.zig

```diff
@@ -62,13 +81,13 @@ test "Opendal BDD test" {
     defer opendal.opendal_metadata_free(&meta);
 
     // The blocking file "test" must have content "Hello, World!"
-    var r: opendal.opendal_result_read = opendal.opendal_operator_blocking_read(bddTest.p, bddTest.path);
+    var r: opendal.opendal_result_read = opendal.opendal_operator_blocking_read(testkit.p, testkit.path);
     defer opendal.opendal_bytes_free(r.data);
     try testing.expect(r.code == opendal.OPENDAL_OK);
-    try testing.expectEqual(std.mem.len(r.data.*.data), std.mem.len(bddTest.content));
+    try testing.expectEqual(std.mem.len(testkit.content), r.data.*.len);
 
     var count: usize = 0;
-    while (count < std.mem.len(r.data.*.data)) : (count += 1) {
-        try testing.expectEqual(bddTest.content[count], r.data.*.data[count]);
+    while (count < r.data.*.len) : (count += 1) {
```

Comment:
Maybe we can change the C struct layout to -

```c
typedef struct opendal_bytes {
  uintptr_t len;
  const uint8_t *data;
} opendal_bytes;
```

cc @Ji-Xinyou

Reply:
And it seems the `data` array does not end with `0` or we overwrite the sentinel with `len` so the original code get a wrong result.

Reply:
For the fields ordering, we can definitely do that.

As for the data array, yes, it does not end with zero because we have len field, which makes the sentinel unnecessary.

Reply:
Do you need the change maybe after this PR? @tisonkun